lib/Doctrine/DBAL/Driver/Connection.php

```diff
@@ -69,6 +69,14 @@ public function exec($statement);
     /**
      * Returns the ID of the last inserted row or sequence value.
      *
+     * If no sequence name is given, "0" is returned if there currently is no last insert ID available
+     * in the current session or in case of an error or if the driver does not support this operation.
```

Comment:
@Majkl578 this is where the error suppression is explicitly defined. It looks dangerous to me that the return value may be different from what's expected because of a runtime issue.